### PR TITLE
Usage-based pricing base support

### PR DESCRIPTION
**Pull request by paolodamico**
Lays the ground work to support usage-based (also called metered or metered-based) pricing.

- Allows creating [usage-based subscriptions](https://stripe.com/docs/billing/subscriptions/metered-billing), this allows us to charge a fee per ingested event, instead of a flat monthly fee. This type of subscription is handled separately from standard subscriptions and is handled differently. The initial process (the checkout session) only creates a session to set up payment information for future billing. After payment information has been set, on the received webhook a usage-based subscription is created.
    - Introduces `is_metered_billing` attribute to `Plan` to identify such plans.
- The PR also allow us to do tiered (or graduated) pricing, in which we charge a different unit cost depending on certain volume steps (example below). This allows us for instance to give the first 5k or 10k events for free, and then charge a fee for each additional event.

- Introduces the `BILLING_TRIAL_DAYS` environment variable to support adding a trial to subscriptions (the integer value represents the number of trial days available in each new subscription). Please note that at this point this will only affect usage-based pricing, **and not flat-fee plans** that already existed before.

In addition, this PR:
- Removes repeated tests.
- Refactors the `create_checkout_session` method for more clarity.
- Refactors the webhook tests to avoid having almost all tests in a single file which makes them hard to read and maintain.

### Additional notes
- Support for monthly flat-fee plans is maintained.
    - The above includes full support for the startup plan.﻿

We will definitely have to test this with a few users before releasing to everyone as there might be some nuances from Stripe we're not considering.

### CODE REVIEW

**Comment by paolodamico** [comment]
CC @timgl in case you want to review

File: multi_tenancy_settings.py [hunk 1]
```diff
@@ -28,3 +28,9 @@
 STRIPE_API_KEY = os.environ.get("STRIPE_API_KEY", "")
 STRIPE_PUBLISHABLE_KEY = os.environ.get("STRIPE_PUBLISHABLE_KEY", "")
 STRIPE_WEBHOOK_SECRET = os.environ.get("STRIPE_WEBHOOK_SECRET", "")
+
+
+# Business rules
+# https://github.com/posthog/posthog-production
```
Comment (Twixes) [inline multi_tenancy_settings.py:34]: What's the link for?

**Review by Twixes — CHANGES_REQUESTED** [review]
Looks good! Only please give the migration a meaningful name.

File: multi_tenancy_settings.py (hunk 1 above)
Comment (paolodamico) [reply multi_tenancy_settings.py:34]: Good catch, documented the `posthog-production` environment variables on the README now.

**Pull request merged**